toolkit/docs/how_it_works/0_intro.md

```diff
@@ -2,23 +2,308 @@ How The Build System Works
+    toolchainChoice -->|no| pullRemote
+    toolchainChoice -->|yes| pullTC 
+    pullRemote --> tcRPMs
+    pullTC -->hydrateTC
+    tcArchiveOld --> pullTC
+    hydrateTC --> tcRPMs
+    buildRawTC --> bsRPMS
+    bsRPMS --> buildTC
+    sources-->buildTC
+    localSpecs --> buildTC
+    buildTC --> tcArchiveNew
+    tcArchiveNew --> pullTC
+    tcRPMs-->tcPopulated
+    tcPopulated --> done
+```
+
+<br/>
+
 ### Package Generation
 Package generation is discussed in detail [here](2_local_packages.md) and [here](3_package_building.md). At a high level, the build system scans the local project directory for `*.spec` files which define the local packages. It then packages these files into `*.src.rpm` files by either including local source files, or downloading matching ones from a source server. The `*.src.rpm` files required for the currently selected configuration are then built in the correct order to resolve all dependencies. The build system will automatically download any build dependencies not satisfied by the local packages.
 
+```mermaid
+flowchart TD
+
+    %% style definitions
+    classDef io fill:#247BA0,stroke:#333,stroke-width:2px,color:#fff;
+    classDef process fill:#B05E2F,stroke:#333,stroke-width:2px,color:#fff;
+    classDef decision fill:#51344D,stroke:#333,stroke-width:2px,color:#fff;
+    classDef goodState fill:#566E40,stroke:#333,stroke-width:2px,color:#fff;
+    classDef badState fill:#BC4B51,stroke:#333,stroke-width:2px,color:#fff;
+
+    %% state nodes
+    start(["Start (make build-packages)"]):::goodState
+    done([Done]):::goodState
+    error[[Error]]:::badState
+
+    %% Rpm nodes
+    tcPopulated([Toolchain populated]):::goodState
+    localSpecs[/Local SPECS/]:::io
+    tcRPMs[(Toolchain RPMs)]:::io
+    createChroot[Create Chroot]:::process
+    chroot[/chroot/]:::io
+    iSRPMs[/Intermediate SRPMs/]:::io
+    parse["Parse Specs (specreader)"]:::process
+    specjson[/"Dependency Data (specs.json)"/]:::io
+    buildGraph[" Build Graph (grapher) "]:::process
+    depGraph[/"Dependency Graph (graph.dot)"/]:::io
+    cacheGraph[/"Cached Graph (cached_graph.dot)"/]:::io
+    sources[/Sources/]:::io
+    packSRPM[Pack SRPM]:::process
+    pkgFetcher["Package fetcher (graphpkgfetcher)"]:::process
+    rpmCache[(RPM cache)]:::io
+    remoteRepo[(remote repo)]:::io
+    missingDep[/RPMs to fill Missing Dependencies/]:::io
+    outRPMS[(RPMs built locally)]:::io
+
+    %% Rpms flow
+    start --> tcPopulated
+    tcPopulated --> packSRPM
+
+    sources-->packSRPM
+    localSpecs-->packSRPM
+    packSRPM-->iSRPMs
+    iSRPMs--> parse
+    parse--> specjson
+    specjson --> buildGraph
+    buildGraph--> depGraph
+    depGraph --> pkgFetcher
+    remoteRepo-->missingDep
+    missingDep-->pkgFetcher
+    pkgFetcher-->rpmCache
+    pkgFetcher-->cacheGraph
+    createChroot-->chroot
+    tcRPMs-->createChroot
+    chroot-...->packSRPM
+    chroot-...->parse
+    chroot-...->pkgFetcher
+    rpmCache-->getDeps
```

Comment:
The block for the cached RPMs is in the way of two arrows (at least in my browser) and somewhat obscures what goes where. Can it be moved somehow?

Reply:
This is a known issue. All other locations seem to add more crossovers